lib/src/utils/video_apis.dart

```diff
@@ -31,7 +31,7 @@ class VideoApis {
     try {
       final response = await _makeRequestHash(videoId, hash);
       final jsonData = jsonDecode(response.body)['request']['files']
-          ['progressive'] as List<dynamic>;
+          ['progressive'] ?? [] as List<dynamic>;
```

Comment:
This will fix the error that we are getting now, But this doesn't fix the actual issue